Gems/AWSCore/Code/Include/Private/ResourceMapping/AWSResourceMappingConstants.h

```diff
@@ -68,7 +68,7 @@ namespace AWSCore
     },
     "AccountIdString": {
         "type": "string",
-        "pattern": "^[0-9]{12}$|EMPTY"
+        "pattern": "^[0-9]{12}$|EMPTY|^$"
```

Comment:
@junbo75 As I remmber EMPTY was added to support a case that there is no account id, as we allow to leave account id as empty string, can we combine these two use cases?

Reply:
I saw that `EMPTY` is used in a default Gem template; I left this case in as acceptable in case there are users out there still operating with this present in their resource mapping file. Otherwise their file would fail to load after this update, right?

Reply:
`EMPTY` was introduced while top-level account id was still required, and project template has to generate a resource mapping file without account id.
But right now, top-level account id is not required, then this EMPTY case is not required anymore to me. You don't need to address it in this PR, but need to confirm whether we can combine it with empty string case, and need a separate jira to track the fix. (as it requires resource mapping tool update as well)

Reply:
I get that, but my concern is that I don't want to start breaking existing users who have already generated a template and may still have `EMPTY` around in their file, because this change means that their file will now fail to load. So unless there exists a mechanism I'm not aware of which will refresh or update an existing resource mapping file to a different default value on update (this doesn't feel like something we'd want to update out from underneath a user, though) then I'm inclined to leave it in until we agree on how to migrate folks or message the change.

But: agreed that this is best addressed as a separate ticket w/ a dedicated discussion, and I also agree we can probably remove `EMPTY` from the template in exchange for setting an empty string.

Reply:
Talked offline, agree with backward incompatible concern
My concern is overcomplicating this field use case which requires special handle in c++ and python, we should optimize it in a separate task.